### ISSUE TITLE
Matrix allows editing messages well in the past

### ISSUE THREAD

**Issue by jeffpaul** [WordPress/matrix#20]
### Expected Behavior

Slack, at least the WP community instance, has a limit of time for when someone can go back and edit their message.  That seems to be on the order of "a couple minutes" as I'm not certain what its _really_ set to.  Matrix should support a similar limit of time for how long a message can be edited.

### Actual Behavior

I was able to edit [this message](https://matrix.to/#/!msRtrhYFZDMTyCufVG:community.wordpress.org/$ENxjpmOzLVT0E6e3hp9MA-4dXdcPRkl4USEqbydv2mQ?via=community.wordpress.org) almost 24 hours after making it which seems like well above the limit of what should be allowed.

**Comment by akirk** [comment]
According to https://github.com/matrix-org/matrix-spec-proposals/issues/1071, it is possible to completely disable redactions (edits) in a room, but not to limit to a certain time threshold.

**Comment by jeffpaul** [comment]
In that case, we should probably disable redactions until there's an option to limit that to a specifc time threshold.

**Comment by akirk** [comment]
Do you know if there had been a wide consensus on limiting the editing threshold in Slack at the time when this was introduced?

**Comment by jeffpaul** [comment]
@akirk no, but that seems like the better default than letting some (possibly for nefarious reasons) edit a message significantly in the past.

**Comment by akirk** [comment]
Actually you can see the edit history when you click the "(edited)" text:
<img width="787" alt="Screenshot 2023-12-06 at 13 41 48" src="https://github.com/WordPress/Matrix/assets/203408/e99dc110-e9e1-49cd-8c31-8362a517fc00">

**Comment by jeffpaul** [comment]
Seeing the edit history as great and all, but the fact that someone could change historical records after a *VERY* short timeframe is extremely sub-optimal.

**Comment by psrpinto** [comment]
@jeffpaul Could you give an example of a situation where someone editing a message poses a problem? 

I'm asking because I'm struggling to understand why the fact that users can edit their own messages poses a problem. I could understand that being an issue if the edit history was not available, but that is not the case, as mentioned above.

**Comment by jeffpaul** [comment]
Let's say I state an opinion on one side of a large discussion / decision to be made (e.g. "Should we migrate from Slack to Matrix?") and then later go back and change that opinion, it would make the history look like I said something wildly different than what I really did.  That's not even really a nefarios one.  You could also see examples where someone harasses some one else and then later goes back to edit/delete those messages so there's no history of that happening; possibly even worse examples of that which would otherwise before being edited/deleted be actual Code of Conduct violations that we'd want to be able to have evidence of.
